platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/oauth/OAuthSecurityImpl.java

```diff
@@ -97,42 +102,52 @@ public OAuthSecurityImpl(TokenStorage tokenStorage) {
   }
 
   /**
-   * Gets the user's access token from the token storage to set it to the OAUTH header.
+   * Gets the user's access token from the token storage to set it to the OAUTH header. Used when a
+   * source is configured to use Authentication Code flow/grant.
    *
    * @param client Non-null client to set the access token on.
-   * @param subject subject used to get the user's id (email or username)
+   * @param subject subject used to get the session ID
    * @param sourceId the id of the source using OAuth needed to get the correct tokens
    */
   public void setUserTokenOnClient(Client client, Subject subject, String sourceId) {
     if (client == null || subject == null || Strings.isBlank(sourceId)) {
       return;
     }
 
-    String userId = SubjectUtils.getEmailAddress(subject);
-    if (userId == null) {
-      userId = SubjectUtils.getName(subject);
+    Session session = subject.getSession();
+    if (session == null) {
+      LOGGER.warn("The user's session is not available.");
+      return;
     }
 
-    TokenInformation.TokenEntry tokenEntry = tokenStorage.read(userId, sourceId);
+    String sessionId = session.getHost();
```

Comment:
Same comment. If there's a shared workstation, the host would be the same. Imo we shouldn't use it as the identifier

Reply:
Like the response above, it's the session ID.